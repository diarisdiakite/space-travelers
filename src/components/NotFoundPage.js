import React, { useEffect } from 'react';
import { useNavigate } from 'react-router-dom';

function NotFoundPage() {
  const navigate = useNavigate();

  useEffect(() => {
    setTimeout(() => {
      navigate('/');
    }, 3000);
  });

  return (
    <div className="main-container">
      <h2>This Page is not found</h2>
      <p>
        Please contact us at
        <strong> diarisdiakite@gmail.com </strong>
        and
        <strong> brunokambere@gmail.com</strong>
      </p>
    </div>
  );
}

export default NotFoundPage;
