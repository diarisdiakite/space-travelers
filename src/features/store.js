import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './missions/missionsSlice';
import usersReducer from './users/usersSlice';
import rocketsSlice from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    users: usersReducer,
    rockets: rocketsSlice,
  },
});

export default store;
