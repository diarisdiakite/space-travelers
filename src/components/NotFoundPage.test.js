import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../features/store';
import NotFoundPage from './NotFoundPage';

describe('NotFoundPage component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <NotFoundPage />
          </MemoryRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
