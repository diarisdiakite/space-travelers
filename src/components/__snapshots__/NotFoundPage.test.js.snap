// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage component renders correctly 1`] = `
<div
  className="main-container"
>
  <h2>
    This Page is not found
  </h2>
  <p>
    Please contact us at
    <strong>
       diarisdiakite@gmail.com 
    </strong>
    and
    <strong>
       brunokambere@gmail.com
    </strong>
  </p>
</div>
`;
